flow/project.py

```diff
@@ -1829,24 +1963,40 @@ def _print_progress(x):
         except KeyError:
             cached_status = dict()
 
-        _get_job_status = functools.partial(self.get_job_status,
-                                            ignore_errors=ignore_errors,
-                                            cached_status=cached_status)
+        _get_job_labels = functools.partial(self._get_job_labels,
+                                            ignore_errors=ignore_errors)
+
+        _get_group_status = functools.partial(self._get_group_status,
+                                              jobs=jobs,
+                                              ignore_errors=ignore_errors,
+                                              cached_status=cached_status)
+
+        groups = []
+        for group in self._groups.values():
+            if len(group.operations) == 1:
+                for name in group.operations:
+                    if name == group.name:
+                        groups.append(group)
+
+        op_results = list(tqdm(
+            iterable=map(_get_group_status, groups),
+            desc="Collecting job status per operation.",
+            total=len(groups)))
```

Comment:
Why aren't we parallelizing this? We could over groups right? I know there aren't as many operations typically as jobs, but we should see some performance improvement using threads considering all the IO.

Reply:
There will be pickling problems for aggregator function. We could however just use `ThreadPool()` in order to parallelize this.

Reply:
We could mention that `process` parallelization only applies to the labels then.

Reply:
@b-butler let's mark this as TODO for now. I think I'll try to parallelize this in #336. If that fails, then I'll add 
> Via 'process' parallelization you can only fetch labels for a job.